### ISSUE TITLE
Focusable A elements cannot be missing an href attribute or explicit role

### ISSUE THREAD

**Issue by dmytro-fedkiv-la** [dmytro-fedkiv-organization/level-ci-github-integration#25]
**Rule description**  
One or more A elements with a tabindex set to 0 or a positive value, excluding those elements with aria-hidden='true' or an href attribute set to a text value, not intentionally hidden in the DOM, lacks a role attribute.

**Tool name**  
Access Engine

**Severity**  
High

**WCAG Success criteria**  
[4.1.2 Name, Role, Value - Level A](https://www.w3.org/WAI/WCAG22/Understanding/name-role-value)

**Why it matters**  
When <code>A</code> elements do not communicate an href attribute or an appropriate explicit role, users of screen readers used by people who are blind or low vision will not know the function of the focusable element. Without knowing the function they may incorrectly use the element or attempt to interact with the element in an unknown manner.  Users of speech recognition may not be able to access the control by voicing the corresponding role which may cause them to not believe the control is interactive.

**What to do**  
The most direct way to communicate the link role of an anchor element is to use the href attribute.  Adding an href will also place the anchor element in the focus order.  If the href cannot be used or if a different role is needed, add an appropriate ARIA <code>role</code> attribute to the element in order to communicate the function of the element.  For example, if the element functions as a button, add <code>role="button"</code>. If not using href or if using a role with a different keyboard pattern from link other requirements apply as well such as adding the correct keyboard pattern support and attributes related to the control (as needed).

**Task in the Level Access Platform**  
https://dmytro-fedkiv-local.localhost-dashboard.essentialaccessibility.com/my-tasks/6a9b245880141057bb1cd763/6a9b249b80141057bb1cddf1

**Comment by dmytro-fedkiv-la** [comment]
<a href="https://dmytro-fedkiv-local.localhost-dashboard.essentialaccessibility.com/tasks/6a9b249b80141057bb1cddf1/attachments/6a9b249b80141057bb1cddef/download" target="_blank" rel="noopener noreferrer">Linked Findings.csv</a>

**Comment by dmytro-fedkiv-la** [comment]
Task pushed to GitHub Issues by Dmytro Fedkiv from Level Access Platform.

Posted by Dmytro Fedkiv via Level Access
